File: bms/go/geo/geo_s2_Cap_Radius/geo_s2_Cap_Radius.go
package test

import (
	"math"
)

// var (
// )

const (
	geo_s1_Radian geo_s1_Angle = 1
)

func (c geo_s2_Cap) Radius() geo_s1_Angle {
	return c.radius.Angle()
}

func (c geo_s1_ChordAngle) Angle() geo_s1_Angle {
	if c < 0 {
		return -1 * geo_s1_Radian
	}
	if c.IsInfinity() {
		return geo_s1_InfAngle()
	}
	return geo_s1_Angle(2 * math.Asin(0.5*math.Sqrt(float64(c))))
}

func (c geo_s1_ChordAngle) IsInfinity() bool {
	return math.IsInf(float64(c), 1)
}

func geo_s1_InfAngle() geo_s1_Angle {
	return geo_s1_Angle(math.Inf(1))
}

type geo_s2_Cap struct {
	radius geo_s1_ChordAngle
}

type geo_s2_Point struct {
	geo_r3_Vector
}

type geo_r3_Vector struct {
}

type geo_s1_ChordAngle float64

type geo_s1_Angle float64

File: translations_only/fail/go/geo/geo_s2_Cap_Radius/claude2/CAPR/geo_s2_Cap_Radius.rs


use std::f64::{self, INFINITY};

struct GeoS2Cap {
    radius: f64,
}

struct GeoS2Point {
    geo_r3_vector: GeoR3Vector,
}

struct GeoR3Vector;

type GeoS1ChordAngle = f64;

type GeoS1Angle = f64;

const GEO_S1_RADIAN: GeoS1Angle = 1.0;

fn radius(c: &GeoS2Cap) -> GeoS1Angle {
    angle(c.radius)
}

fn angle(c: GeoS1ChordAngle) -> GeoS1Angle {
    if c < 0.0 {
        -1.0 * GEO_S1_RADIAN
    } else if is_infinity(c) {
        geo_s1_inf_angle()
    } else {
        2.0 * f64::asin(0.5 * f64::sqrt(c as f64))
    }
}

fn is_infinity(c: GeoS1ChordAngle) -> bool {
    f64::is_infinite(c)
}   

fn geo_s1_inf_angle() -> GeoS1Angle {
    INFINITY
}

